vm-presets.md

```diff
@@ -0,0 +1,185 @@
+Virtual Machine Presets
+=============================
+
+`VirtualMachinePresets` are an extension to general `VirtualMachine`
+Configuration based heavily on `PodPresets` from Kubernetes. When a
+`VirtualMachine` is created, any applicable `VirtualMachinePresets`
+will be applied to the existing spec for the `VirtualMachine`. This allows
+for re-use of common settings that should apply to multiple `VirtualMachines`.
+
+
+Implementation
+------------------------
+
+`VirtualMachinePresets` are implemented as a Kubernetes `Initializer`. This
+means the `VirtualMachine` resource is modified before it is visible to or
+processed by any other component of KubeVirt.
+
+One noteworthy difference between the behavior of a `VirtualMachinePreset` and
+a `PodPreset` is that if an error occurs during processing of a `PodPreset`,
+Kubernetes will continue to create the `Pod` without applying the `PodPreset`.
+KubeVirt will move the VirtualMachine to the `Failed` state instead. The
```

Comment:
As you write it ourself the vm to Failed is something different than what the PodPreset does with a Pod. I would expect that we don't add the VirtualMachinePreset annotation to the vm, log an error event which we can see with `kubectl describe vm myvm` and continue with the next preset.